http3/capsule_type.go

```diff
@@ -0,0 +1,30 @@
+package http3
+
+import "fmt"
+
+const (
+	CapsuleTypeRegisterDatagramContext   CapsuleType = 0x00
+	CapsuleTypeCloseDatagramContext      CapsuleType = 0x01
+	CapsuleTypeDatagram                  CapsuleType = 0x02
+	CapsuleTypeRegisterDatagramNoContext CapsuleType = 0x03
+)
+
+// A CapsuleType represents an HTTP capsule type.
+// https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-03.html#name-capsule-types
+type CapsuleType uint64
```

Comment:
This doesn't seem to be used anywhere, is it?

Reply:
`CapsuleType` and `ContextExtensionType` are intended to support the [HTTP/3 MASQUE datagram draft](https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-03.html), which the WebTransport draft is converging against.

This can be deferred to a future PR.